api/controllers/Irc_EntryController.js

```diff
@@ -63,32 +57,74 @@ module.exports = {
   },
 
   process_event: function (request_body) {
-    if (request_body.operation === 'check in') {
-      return this.saveDetainee(request_body)
-        .then(this.saveEvent.bind(this, request_body));
+    switch (request_body.operation) {
+    case Event.OPERATION_CHECK_IN:
+    case Event.OPERATION_CHECK_OUT:
+    case Event.OPERATION_INTER_SITE_TRANSFER:
```

Comment:
your code looks like you'd be able to handle an update operation, but you'll bounce it here?